xarray/core/variable.py

```diff
@@ -1094,54 +1105,92 @@ def chunk(
+                    data = indexing.ImplicitToExplicitIndexingAdapter(
+                        data, indexing.OuterIndexer
+                    )
 
-        data = self._data
-        if is_duck_dask_array(data):
-            data = data.rechunk(chunks)
-        else:
-            if isinstance(data, indexing.ExplicitlyIndexed):
-                # Unambiguously handle array storage backends (like NetCDF4 and h5py)
-                # that can't handle general array indexing. For example, in netCDF4 you
-                # can do "outer" indexing along two dimensions independent, which works
-                # differently from how NumPy handles it.
-                # da.from_array works by using lazy indexing with a tuple of slices.
-                # Using OuterIndexer is a pragmatic choice: dask does not yet handle
-                # different indexing types in an explicit way:
-                # https://github.com/dask/dask/issues/2883
-                data = indexing.ImplicitToExplicitIndexingAdapter(
-                    data, indexing.OuterIndexer
+                    # All of our lazily loaded backend array classes should use NumPy
+                    # array operations.
+                    kwargs = {"meta": np.ndarray}
+                else:
+                    kwargs = {}
+
+                if utils.is_dict_like(chunks):
+                    chunks = tuple(chunks.get(n, s) for n, s in enumerate(self.shape))
+
+                data = da.from_array(
+                    data,
+                    chunks,
+                    name=name,
+                    lock=lock,
+                    inline_array=inline_array,
+                    **kwargs,
                 )
 
-                # All of our lazily loaded backend array classes should use NumPy
-                # array operations.
-                kwargs = {"meta": np.ndarray}
+        elif manager == "cubed":
+            import cubed  # type: ignore
+
+            if isinstance(chunks, (float, str, int, tuple, list)):
+                raise TypeError  # unsure if cubed.from_array can handle this directly
             else:
-                kwargs = {}
+                chunks = either_dict_or_kwargs(chunks, chunks_kwargs, "chunk")
 
             if utils.is_dict_like(chunks):
-                chunks = tuple(chunks.get(n, s) for n, s in enumerate(self.shape))
+                chunks = {
+                    self.get_axis_num(dim): chunk for dim, chunk in chunks.items()
+                }
 
-            data = da.from_array(
-                data, chunks, name=name, lock=lock, inline_array=inline_array, **kwargs
+            data = self._data
+            if isinstance(data, cubed.Array):
+                data = data.rechunk(chunks)
+            elif is_duck_dask_array(data):
+                raise TypeError("Trying to rechunk a dask array using cubed")
+
+            data = cubed.from_array(
+                data,
+                chunks,
+                spec=spec,
```

Comment:
Minor comment: Since cubed is using the same api (just different kwargs) we should be able to consolidate the code a bit under a `if manager in ("cubed", "dask")` block. Not sure if it will become more readable though.

Reply:
I did start doing that but I found the special casing more confusing than repeated code. This is easy to change later though.